Packages/Status/Sources/Status/Row/StatusRowViewModel.swift

```diff
@@ -62,6 +67,14 @@ public class StatusRowViewModel: ObservableObject {
 
     isFiltered = filter != nil
   }
+    
+  func markSeen() {
+    // called in on appear so we can cache that the status has been seen.
+    if !seen {
+      ReblogCache.shared.cache(status, seen: true)
```

Comment:
You should wrap that in a background DispatchQueue, that could definitely hang the UI.

Reply:
Should you cache only if this a reblog?

Reply:
The cache tests if it's a reblog so it does get checked, can move the check out here if you prefer.